### ISSUE TITLE
Zelda Dungeon Evolution: Fitness to Encourage Backtracking

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEAT#434]
Zelda Dungeon evolution options currently available focus on the length of the A* search path and/or the number of rooms in that path. However, a way to make dungeons more interesting is to force some amount of backtracking (caused by keys or the raft item). So, make a fitness function that specifically encourages A* paths that revisit rooms multiple times.

**Comment by schrum2** [comment]
This will be your first "real" issue. Wait until SCOPE starts to really look at it. Ask me questions frequently to stay on track, and log what you learn in this issue thread.

You should know from your previous issue how the fitness for the number of rooms in the path is calculated. This new fitness function should be the number of rooms that are visited twice along the path. So, the player has to leave the room and then return to it for the score to increase by one.

**Comment by cappsb** [comment]
Allowed for tracking of the exited rooms as well as the rooms that have been traversed more than once. However, the Direct2GAN will never produce a dungeon that requires backtracking

**Comment by schrum2** [comment]
CPPN2GAN also does not generally produce rooms that require backtracking. Tried with commit 3efb39cf9373448504c25e6c69776f186e6d3b73

This isn't too surprising though. Maybe if we evolved long enough, we would see some rooms with backtracking. However, it would be better to rely on things like issue #448 and #446 to encourage backtracking. This issue seems resolved, but I want to keep it open until it is tested.
